src/assets.psm1

```diff
@@ -2294,47 +2298,49 @@ New-SafeguardAssetImportTemplate 'C:\tmp\template.csv' -DisplayName -Description
-    if ($PSBoundParameters.ContainsKey("Port")) { $local:Headers = $local:Headers + ',"Port"' }
-    if ($PSBoundParameters.ContainsKey("ServiceAccountDomainName")) { $local:Headers = $local:Headers + ',"ServiceAccountDomainName"' }
-    if ($PSBoundParameters.ContainsKey("ServiceAccountName")) { $local:Headers = $local:Headers + ',"ServiceAccountName"' }
-    if ($PSBoundParameters.ContainsKey("ServiceAccountPassword")) { $local:Headers = $local:Headers + ',"ServiceAccountPassword"' }
-    if ($PSBoundParameters.ContainsKey("ServiceAccountCredentialType")) { $local:Headers = $local:Headers + ',"ServiceAccountCredentialType"' }
-    if ($PSBoundParameters.ContainsKey("ServiceAccountSecretKey")) { $local:Headers = $local:Headers + ',"ServiceAccountSecretKey"' }
-    if ($PSBoundParameters.ContainsKey("ServiceAccountDistinguishedName")) { $local:Headers = $local:Headers + ',"ServiceAccountDistinguishedName"' }
-    if ($PSBoundParameters.ContainsKey("PrivilegeElevationCommand")) { $local:Headers = $local:Headers + ',"PrivilegeElevationCommand"' }
+    if ($PSBoundParameters.ContainsKey("Description") -or $PSBoundParameters.ContainsKey("All")) { $local:Headers = $local:Headers + ',"Description"' }
+    if ($PSBoundParameters.ContainsKey("AssetPartition") -or $PSBoundParameters.ContainsKey("All")) { $local:Headers = $local:Headers + ',"AssetPartition"' }
+    if ($PSBoundParameters.ContainsKey("NetworkAddress") -or $PSBoundParameters.ContainsKey("All")) { $local:Headers = $local:Headers + ',"NetworkAddress"' }
+    if ($PSBoundParameters.ContainsKey("Port") -or $PSBoundParameters.ContainsKey("All")) { $local:Headers = $local:Headers + ',"Port"' }
+    if ($PSBoundParameters.ContainsKey("ServiceAccountDomainName") -or $PSBoundParameters.ContainsKey("All")) { $local:Headers = $local:Headers + ',"ServiceAccountDomainName"' }
+    if ($PSBoundParameters.ContainsKey("ServiceAccountName") -or $PSBoundParameters.ContainsKey("All")) { $local:Headers = $local:Headers + ',"ServiceAccountName"' }
+    if ($PSBoundParameters.ContainsKey("ServiceAccountPassword") -or $PSBoundParameters.ContainsKey("All")) { $local:Headers = $local:Headers + ',"ServiceAccountPassword"' }
+    if ($PSBoundParameters.ContainsKey("ServiceAccountCredentialType") -or $PSBoundParameters.ContainsKey("All")) { $local:Headers = $local:Headers + ',"ServiceAccountCredentialType"' }
+    if ($PSBoundParameters.ContainsKey("ServiceAccountSecretKey") -or $PSBoundParameters.ContainsKey("All")) { $local:Headers = $local:Headers + ',"ServiceAccountSecretKey"' }
+    if ($PSBoundParameters.ContainsKey("ServiceAccountDistinguishedName") -or $PSBoundParameters.ContainsKey("All")) { $local:Headers = $local:Headers + ',"ServiceAccountDistinguishedName"' }
+    if ($PSBoundParameters.ContainsKey("PrivilegeElevationCommand") -or $PSBoundParameters.ContainsKey("All")) { $local:Headers = $local:Headers + ',"PrivilegeElevationCommand"' }
 
```

Comment:
I'd split this as

if ($PSBoundParameters.ContainsKey("All")) {
      ...  $local:Headers = $local:Headers + .... alll individual paramters ...
}
else {
     .... check for each individual parameter
}

same for the other commands below

Reply:
I did it like that originally but the giant string that had all the headers was difficult to read and identify any typos.  Also you will have to change multiple locations if you want a header changed.  This was just cleaner and easier to identify any issues and make changes.